File: perl/basics/perl-projects/mysql_crud.pl
#!/usr/bin/env perl
# This program will demonstrate crud operation on mysql db
# prerequisite
# $ sudo apt install mysql-server
# $ sudo apt install libmysqlclient-dev
# $ sudo perl -MCPAN -e shell
# cpan> install DBI
# cpan[2]> install DBD::mysql
# Strict and warnings are recommended.
use strict;
use warnings;
use DBI;

# Connect to the database.
my $dbh = DBI->connect( "DBI:mysql:database=test;host=localhost",
  "root", "toor", { 'RaiseError' => 1 } );

# Drop table 'foo'. This may fail, if 'foo' doesn't exist
# Thus we put an eval around it.
eval { $dbh->do("DROP TABLE foo") };
print "Dropping foo failed: $@\n" if $@;

# Create a new table 'foo'. This must not fail, thus we don't
# catch errors.
$dbh->do("CREATE TABLE foo (id INTEGER, name VARCHAR(20))");

# INSERT some data into 'foo'. We are using $dbh->quote() for
# quoting the name.
$dbh->do( "INSERT INTO foo VALUES (1, " . $dbh->quote("Tim") . ")" );

# same thing, but using placeholders (recommended!)
$dbh->do( "INSERT INTO foo VALUES (?, ?)", undef, 2, "Jochen" );

# now retrieve data from the table.
my $sth = $dbh->prepare("SELECT * FROM foo");
$sth->execute();
while ( my $ref = $sth->fetchrow_hashref() ) {
  print "Found a row: id = $ref->{'id'}, name = $ref->{'name'}\n";
}
$sth->finish();

# Disconnect from the database.
$dbh->disconnect();

File: python/basics/python2-projects/mysql/mysql_crud.py
#!/usr/bin/env python
# Requires Python 2+
# prerequisite
# sudo apt install python-pip python-dev libmysqlclient-dev
# upgrade package manager
# pip install --upgrade pip
# Install MySQL-python package
# pip install MySQL-python
# or
# sudo apt install python-mysqldb
# My sql Crud operations using python
# Usage
# python mysql_crud.py
import MySQLdb

# Instantiate connection object and connect to MySQL database
conn = MySQLdb.connect(host="localhost", user="root", passwd="toor", db="test")

# Instantiate cursor object
cursor = conn.cursor()

# Drop Table
cursor.execute("DROP TABLE IF EXISTS TableTest")

# Create Table
cursor.execute("CREATE TABLE TableTest(Id INT PRIMARY KEY AUTO_INCREMENT, \
             Name VARCHAR(25))")
# Insert Some Data
cursor.execute("INSERT INTO TableTest(Name) VALUES('TestData1')")
cursor.execute("INSERT INTO TableTest(Name) VALUES('TestData2')")
cursor.execute("INSERT INTO TableTest(Name) VALUES('TestData3')")
cursor.execute("INSERT INTO TableTest(Name) VALUES('TestData4')")
cursor.execute("INSERT INTO TableTest(Name) VALUES('TestData5')")
conn.commit()

# Retrieve Data
# execute SQL select statement
cursor.execute("SELECT Id,Name FROM TableTest")

rows = cursor.fetchall()

# Iterate
# get and display one row at a time.
for row in rows:
    # data from rows
    print row

# Update Table
cursor.execute("UPDATE TableTest SET Name = %s WHERE Id = %s",
        ("TestData41", "4"))

print "Number of rows updated:", cursor.rowcount

# Delete
cursor.execute("DELETE FROM TableTest WHERE Id = %s", "2")

print "Number of rows deleted:", cursor.rowcount

# Close cursor
cursor.close ()
# Close database connection
conn.close()
